typing/jkind.ml

```diff
@@ -972,41 +972,82 @@ module Const = struct
     (* CR layouts v3: change to [Maybe_null] when separability is implemented. *)
     let float64 =
       { jkind =
-          mk_jkind (Base Float64) ~mode_crossing:true ~nullability:Non_null;
+          mk_jkind (Base Float64) ~mode_crossing:false ~nullability:Non_null;
         name = "float64"
       }
 
+    (* CR layouts v3: change to [Maybe_null] when separability is implemented. *)
+    let kind_of_unboxed_float =
+      { jkind =
+          mk_jkind (Base Float64) ~mode_crossing:true ~nullability:Non_null;
+        name = "float64 mod everything"
```

Comment:
I don't think names like this are the right thing to do here; these are used in printing verbatim, and the usual thing to do when printing `<layout> mod <modes>` is to put a break hint instead of spaces.

It kinda seems like we might want to not use Builtin for these (the `mod everything` kinds) and pull up a level of abstraction, so we avoid weirdness in printing.

Reply:
I think you've made two points:
* It's not obvious these definitions belong here. I thought about putting them elsewhere, but concluded they want to be in the `all` list so that `subst` memoizes them appropriately. Open to other suggestions, though.
* The names are weird.  Agreed!  But they do sometimes get printed, and this means they get printed as things that parse.  The lack of line breaks is sad, but I'm not sure I see a better alternative short of a larger refactoring of `Builtin`?

Reply:
I think we should not try to make this great here. `Builtin` was a cheap-and-cheerful (well not so cheerful) approach from its birth, and is mostly sad. It should be removed. So I think if the name with spaces works ok in practice, it's fine for now.